From 40112957ca6b830e6299a2c1358049bc8a667557 Mon Sep 17 00:00:00 2001
From: Michal Sojka <michal.sojka@cvut.cz>
Date: Wed, 21 Aug 2024 08:32:49 +0200
Subject: [PATCH] Make next-matching-window behave as I want

The next-matching-window command always cycles between the matching
windows, independently on whether any of the window is focused or not.

I want the cycling to happen only when any window is focused. If not,
the LRU window should be focuses, i.e. the same window as before, i.e.
no cycling.
---
 swayr/src/cmds.rs | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/swayr/src/cmds.rs b/swayr/src/cmds.rs
index 18fa450..8e1fb4c 100644
--- a/swayr/src/cmds.rs
+++ b/swayr/src/cmds.rs
@@ -1589,13 +1589,12 @@ fn focus_window_in_direction_1(
         lru_a.cmp(&lru_b).reverse()
     });
 
+    if !wins.iter().any(|w| w.node.focused) {
+        return focus_window_by_id(wins.first().unwrap().node.id);
+    }
+
     let is_focused_window: Box<dyn Fn(&t::DisplayNode) -> bool> =
-        if !wins.iter().any(|w| w.node.focused) {
-            let last_focused_win_id = wins.first().unwrap().node.id;
-            Box::new(move |dn| dn.node.id == last_focused_win_id)
-        } else {
-            Box::new(|dn| dn.node.focused)
-        };
+        Box::new(|dn| dn.node.focused);
 
     let mut iter: Box<dyn Iterator<Item = &&t::DisplayNode>> = match dir {
         Direction::Forward => Box::new(wins.iter().rev().cycle()),
-- 
2.45.2

